testah-junit/src/main/java/org/testah/framework/testPlan/TestSystem.java

```diff
@@ -0,0 +1,608 @@
+        if (null == testCase) {
+            testCase = new ThreadLocal<>();
+        }
+        return testCase;
+    }
+
+    /**
+     * Gets the test case.
+     *
+     * @return the test case
+     */
+    public TestCaseDto getTestCase() {
+        return getTestCaseThreadLocal().get();
+    }
+
+    /**
+     * filterTest to see what should run.
+     * @param description junit 4 style Description.
+     */
+    public void filterTest(final Description description) {
+        filterTest(description,
+            TestPlanAnnotationDto.create(description.getTestClass().getAnnotation(TestPlan.class)),
+            TestCaseAnnotationDto.create(description.getAnnotation(TestCase.class)));
+    }
+
+    /**
+     * filterTest test to run.
+     *
+     * @param description the description
+     * @param testPlan    testplan info from the annotation
+     * @param testCase    testcase info from the annotation
+     */
+    public void filterTest(final Description description, TestPlanAnnotationDto testPlan, TestCaseAnnotationDto testCase) {
+        final String name = description.getDisplayName();
+        final KnownProblem kp = description.getAnnotation(KnownProblem.class);
+        TestCaseDto test = new TestCaseDto();
+        test = TestDtoHelper.fill(test, testCase, kp, testPlan);
+        if (!getTestFilter().filterTestCase(test, name)) {
+            addIgnoredTest(name, "METADATA_FILTER");
+            Assume.assumeTrue("Filtered out, For details use Trace level logging" +
+                "\nCheck your filter settings in Testah.properties for " +
+                "filter_DEFAULT_filterIgnoreKnownProblem", false);
+        }
+
+        if (null != TS.params().getFilterIgnoreKnownProblem()) {
+            if (null != kp) {
+                if ("true".equalsIgnoreCase(TS.params().getFilterIgnoreKnownProblem())) {
+                    addIgnoredTest(name, "KNOWN_PROBLEM_FILTER");
+                    Assume.assumeTrue("Filtered out, KnownProblem found: " + kp.description() +
+                        "\nCheck your filter settings in Testah.properties for " +
+                        "filter_DEFAULT_filterIgnoreKnownProblem", false);
+                }
+            } else if ("false".equalsIgnoreCase(TS.params().getFilterIgnoreKnownProblem())) {
+                addIgnoredTest(name, "KNOWN_PROBLEM_FILTER");
+                Assume.assumeTrue(
+                    "Filtered out, KnownProblem Not found and is required\nCheck your filter" +
+                        " settings in Testah.properties for filter_DEFAULT_filterIgnoreKnownProblem",
+                    false);
+            }
+        }
+    }
+
+    /**
+     * Start test plan.
+     *
+     * @param desc        the desc
+     * @param testPlan    the test plan
+     * @param knowProblem the know problem
+     * @return the test plan dto
+     */
+    public TestPlanDto startTestPlan(final Description desc, final TestPlanAnnotationDto testPlan, final KnownProblem knowProblem) {
```

Comment:
this is confusing: testPlan for both a TestPlanAnnotationDtp and a TestPlanDto.

Reply:
In order to support Junit 4 and 5 annotations with the same code I basically created a metadata holder that 100% lines up to the annotations, where as TestPlanDto does not, the goal is the same, get the info from the annotation, and put it into a testplan dto.  The issue is trying to do that with 2 distinct annotations at are pretty much identical but different.  This is the approach that worked, I agree, would like a better solution, might refactor when have more time.